ros2_control_demo_hardware/src/rrbot_system_position_only.cpp

```diff
@@ -0,0 +1,217 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ros2_control_demo_hardware/rrbot_system_position_only.hpp"
+
+#include <chrono>
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace ros2_control_demo_hardware
+{
+
+return_type RRBotSystemPositionOnlyHardware::configure(
+  const hardware_interface::HardwareInfo & info)
+{
+  if (configure_default(info) != return_type::OK) {
+    return return_type::ERROR;
+  }
+
+  hw_start_sec_ = stod(info_.hardware_parameters["example_param_hw_start_duration_sec"]);
+  hw_stop_sec_ = stod(info_.hardware_parameters["example_param_hw_stop_duration_sec"]);
+  hw_slowdown_ = stod(info_.hardware_parameters["example_param_hw_slowdown"]);
+  hw_states_.resize(info_.joints.size());
+  hw_commands_.resize(info_.joints.size());
+  for (uint i = 0; i < info_.joints.size(); i++) {
+    hw_states_[i] = std::numeric_limits<double>::quiet_NaN();
+    hw_commands_[i] = std::numeric_limits<double>::quiet_NaN();
+  }
+
+  // TODO(anyone): we should provide helpers for this checking somewhere
+  {
+    for (const hardware_interface::ComponentInfo & joint : info_.joints) {
+      // RRBotSystemPositionOnly has exactly one state and command interface on each joint
+      if (joint.command_interfaces.size() != 1) {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSystemPositionOnlyHardware"),
+          "Joint '%s' has %d command interfaces found. 1 expected.",
+          joint.name.c_str(), joint.command_interfaces.size());
+        return return_type::ERROR;
+      }
+
+      if (joint.command_interfaces[0].name != hardware_interface::HW_IF_POSITION) {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSystemPositionOnlyHardware"),
+          "Joint '%s' have %s command interfaces found. '%s' expected.",
+          joint.name.c_str(), joint.command_interfaces[0].name.c_str(),
+          hardware_interface::HW_IF_POSITION);
+        return return_type::ERROR;
+      }
+
+      if (joint.state_interfaces.size() != 1) {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSystemPositionOnlyHardware"),
+          "Joint '%s' has %d state interface. 1 expected.",
+          joint.name.c_str(), joint.state_interfaces.size());
+        return return_type::ERROR;
+      }
+
+      if (joint.state_interfaces[0].name != hardware_interface::HW_IF_POSITION) {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSystemPositionOnlyHardware"),
+          "Joint '%s' have %s state interface. '%s' expected.",
+          joint.name.c_str(), joint.state_interfaces[0].name.c_str(),
+          hardware_interface::HW_IF_POSITION);
+        return return_type::ERROR;
+      }
+    }
+  }
+
+  // TODO(all): ada support for different order of joints than "joint1", "joint2",...
```

Comment:
```suggestion
  // TODO(all): add support for different order of joints than "joint1", "joint2",...
```

Reply:
I'll remove the comment, the joints should be exported in defined order as in URDF. We should add test for it.